docker/images.go

```diff
@@ -39,26 +39,23 @@ func (d *docker) ImageRemove(ctx context.Context, imageID string, force, pruneCh
 }
 
 func (d *docker) getImageStatus(ctx context.Context, image types.ImageSummary) (status string) {
-	if len(image.RepoTags) == 0 {
+	if image.RepoTags[0] == "<none>:<none>" {
 		status = string(entities.ImageStatusUnUsedDangling)
+		return
 	}
 
-	containers, _ := d.cli.ContainerList(ctx, types.ContainerListOptions{})
-	newImgs := make(map[string][]string)
-	for _, cont := range containers {
-		if cont.ImageID != image.ID {
-			status = string(entities.ImageStatusUnUsed)
-		} else {
-			newSlice := newImgs[image.ID]
-			if newSlice == nil {
-				newSlice = make([]string, 0)
-			}
-			newSlice = append(newSlice, cont.ID)
-			newImgs[image.ID] = newSlice
+	if image.Containers == 0 {
```

Comment:
According to the docs:
> Number of containers .... **This size is not calculated by default, and depends on which API endpoint is used. `-1` indicates that the value has not been set / calculated.**

So this number will always be "-1" cause we didn't set it.

But if you're still sure that I might be wrong, let's discuss it in a meeting or something.

Reply:
```go
type ImageSummary struct {

	// Number of containers using this image. Includes both stopped and running
	// containers.
	//
	// This size is not calculated by default, and depends on which API endpoint
	// is used. `-1` indicates that the value has not been set / calculated.
	//
	// Required: true
	Containers [int64](https://pkg.go.dev/builtin#int64) `json:"Containers"`
```

yes you  are rigth. `-1` means no api still set/calc the number of containers at this moment